jOOQ/src/main/java/org/jooq/tools/Convert.java

```diff
@@ -1219,14 +1219,16 @@ private static final String patchIso8601Time(String string) {
                  : string;
         }
 
-        private static final String patchIso8601Timestamp(String string, boolean t) {
-            if (string.length() > 11)
-                if (t && string.charAt(10) == ' ')
-                    return string.substring(0, 10) + "T" + string.substring(11);
-                else if (!t && string.charAt(10) == 'T')
-                    return string.substring(0, 10) + " " + string.substring(11);
-
-            return string;
+        private static String patchIso8601Timestamp(String string, boolean t) {
+            if (t) {
+                if (string.contains("T")) {
+                    return string;
+                } else {
+                    return string.replaceFirst(" ", "T");
```

Comment:
This is a regular expression replacement and despite there being an optimisation for expressions that are "obviously" not regular expressions, there is an `int[]` allocation in many JDK versions (haven't checked 17 yet), which is very costly.